cdap-docs/reference-manual/source/release-notes.rst

```diff
@@ -154,6 +154,16 @@ Known Issues
     
   All versions of CDAP running Twill version 0.4.0 with this configuration can exhibit this
   problem (`TWILL-110 <https://issues.apache.org/jira/browse/TWILL-110>`__).
+- Typically Datasets are bundled as part of Applications. When an Application is upgraded and redeployed,
+  any changes in Datasets will not be redeployed. This is because Datasets can be shared across applications,
+  and an incompatible schema change can break other applications that are using the Dataset.
+  A workaround (`CDAP-1253 <https://issues.cask.co/browse/CDAP-1253>`__) is to set a configuration,
+  "dataset.unchecked.upgrade", that always redeploys Datasets when an Application is redeployed.
+  When this configuration is set, the recommended process to deploy a upgraded Dataset is to first stop
+  all Applications that are using this dataset before deploying the Application.
+  This lets all containers (Flows, Services, etc) to pick up the new Dataset changes.
+  However, this is an unsafe operation since no schema validation is done and the upgraded Dataset may
```

Comment:
When Datasets are upgraded using `dataset.unchecked.upgrade`, no schema compatibility checks are performed by the system. Hence it is very important that the developer verifies the new changes are backwards-compatible, and make sure that other applications that are using the Dataset can work with the new changes.